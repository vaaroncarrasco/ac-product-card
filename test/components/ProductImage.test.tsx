import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/product';

describe('ProductTitle', () => {

  test('should match snapshot with img', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://hola.jpg" />
    );

    expect( wrapper.toJSON() ).toMatchSnapshot();

  });
  
  test('should display comp with no image', () => {
    const wrapper = renderer.create(
      <ProductCard product={ product2 }>
        {
          () => (
            <ProductImage />
          )
        }
      </ProductCard>
    );

    expect( wrapper.toJSON() ).toMatchSnapshot();

  });


});
