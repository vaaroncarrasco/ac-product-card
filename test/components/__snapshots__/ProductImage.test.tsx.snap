// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should display comp with no image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffee mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductTitle should match snapshot with img 1`] = `
<img
  alt="Coffee mug"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
